vllm/model_executor/layers/quantization/awq_triton.py

```diff
@@ -43,6 +43,9 @@ def awq_dequantize_kernel(
 
     # Load the weights.
     iweights = tl.load(qweight_ptr + offsets, masks)
+    iweights = tl.interleave(iweights, iweights)
+    iweights = tl.interleave(iweights, iweights)
+    iweights = tl.interleave(iweights, iweights)
```

Comment:
Can you just broadcast this?  I would assume it would be more efficient than using 3 interleaves.

Reply:
Yes, it's possible.  I just tried:

iweights = tl.broadcast_to(iweights[:, :, None],
                               (BLOCK_SIZE_Y, BLOCK_SIZE_X, 8)).reshape(
                                   BLOCK_SIZE_Y, BLOCK_SIZE_X * 8)
                                 
However, it's slightly slower, especially on the h100, performance drops from 27k tok/s to 24k tok/s, a roughly 10% drop in performance.  I tried on MI300 and also noticed drop in performance,  9898 tok/s to 9821 tok/s, not as much, but it's had to tell with MI300 since Triton performance is already poor.  I would try on MI250, but it's being hammered at the moment by CI jobs.

I'm going to revert back to the interleave based implementation.

Reply:
Maybe if it's faster in the future, it could be used.  Maybe make a note of it in code? @hegemanjw4amd